airflow/gcp/operators/campaign_manager.py

```diff
@@ -0,0 +1,382 @@
+    def _set_bucket_name(name: str) -> str:
+        bucket = name if not name.startswith("gs://") else name[5:]
+        return bucket.strip("/")
+
+    def execute(self, context: Dict):
+        hook = GoogleCampaignManagerHook(
+            gcp_conn_id=self.gcp_conn_id,
+            delegate_to=self.delegate_to,
+            api_version=self.api_version,
+        )
+        gcs_hook = GoogleCloudStorageHook(
+            google_cloud_storage_conn_id=self.gcp_conn_id, delegate_to=self.delegate_to
+        )
+        # Get name of the report
+        report = hook.get_report(
+            file_id=self.file_id, profile_id=self.profile_id, report_id=self.report_id
+        )
+        report_name = self.report_name or report.get(
+            "fileName", str(uuid.uuid4())
+        )
+        report_name = self._set_report_name(report_name)
+
+        # Download the report
+        self.log.info("Starting downloading report %s", self.report_id)
+        request = hook.get_report_file(
+            profile_id=self.profile_id, report_id=self.report_id, file_id=self.file_id
+        )
+        with tempfile.NamedTemporaryFile() as temp_file:
+            downloader = http.MediaIoBaseDownload(
+                fd=temp_file, request=request, chunksize=self.chunk_size
+            )
+            download_finished = False
+            while not download_finished:
+                _, download_finished = downloader.next_chunk()
+
+            temp_file.flush()
+            # Upload the local file to bucket
+            gcs_hook.upload(
+                bucket_name=self.bucket_name,
+                object_name=report_name,
+                gzip=self.gzip,
+                filename=temp_file.name,
+                mime_type="text/csv",
+            )
+
+        self.xcom_push(context, key="report_name", value=report_name)
+
+
+class GoogleCampaignManagerInsertReportOperator(BaseOperator):
+    """
+    Creates a report.
+
+    .. seealso::
+        Check official API docs:
+        https://developers.google.com/doubleclick-advertisers/v3.3/reports/insert
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:GoogleCampaignManagerInsertReportOperator`
+
+    :param profile_id: The DFA user profile ID.
+    :type profile_id: str
+    :param report: Report to be created.
+    :type report: Dict[str, Any]
+    :param api_version: The version of the api that will be requested for example 'v3'.
+    :type api_version: str
+    :param gcp_conn_id: The connection ID to use when fetching connection info.
+    :type gcp_conn_id: str
+    :param delegate_to: The account to impersonate, if any. For this to work, the service accountmaking the
+        request must have  domain-wide delegation enabled.
+    :type delegate_to: str
+    """
+
+    template_fields = (
+        "profile_id",
+        "report",
+        "api_version",
+        "gcp_conn_id",
+        "delegate_to",
+    )
+
```

Comment:
We should add json as a template_ext to support creating reports from *.json files.